src/main/kotlin/no/item/xp/plugin/util/GenerateInterfaceName.kt

```diff
@@ -0,0 +1,22 @@
+package no.item.xp.plugin.util
+
+import java.io.File
+
+@Suppress("FunctionName")
+fun GenerateInterfaceName(fileName: File): String? {
+  val filePath: String? = pathComponent(fileName.absolutePath)
+  val newFileName: String = filePath + "\\" + fileName.nameWithoutExtension + ".ts"
+  val file = File(newFileName)
+  val isNewFileCreated: Boolean = file.createNewFile()
+  return if (isNewFileCreated) {
+    newFileName
+  } else {
+    println("$newFileName already exists.")
+    null
+  }
+}
+
+fun pathComponent(filename: String): String? {
```

Comment:
Is the return type nullable here?

Reply:
No, changed to a strict String return type